### ISSUE TITLE
Getting TypeError function takes at most 2 arguments (3 given)

### ISSUE THREAD

**Issue by balandongiv** [wweir827/CHAMP#9]
Interesting work by the author. I am not sure whether support is still provided for this package.

When running the script [test_champ.py](https://github.com/wweir827/CHAMP/blob/master/test/test_champ.py), the compiler return an error
`
Getting TypeError function takes at most 2 arguments (3 given)`

The full traceback is as below

```
  0%|          | 0/10 [00:00<?, ?it/s]
0it [00:00, ?it/s]
  0%|          | 0/10 [00:00<?, ?it/s]
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/site-packages/champ/leiden_ext.py", line 128, in _run_leiden_parallel
    attribute=attribute,output_dictionary=True,niterations=niterations)
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/site-packages/champ/leiden_ext.py", line 58, in run_leiden
    g = ig.Graph.Read_GraphMLz(gfile)
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/site-packages/igraph/__init__.py", line 2315, in Read_GraphMLz
    return cls.Read_GraphML(tmpfile, directed=directed, index=index)
TypeError: function takes at most 2 arguments (3 given)
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/rpb/PycharmProjects/champ_test/test_champ.py", line 325, in <module>
    main()
  File "/home/rpb/PycharmProjects/champ_test/test_champ.py", line 321, in main
    test_parallel_run()
  File "/home/rpb/PycharmProjects/champ_test/test_champ.py", line 30, in test_parallel_run
    progress=True,calc_sim_mat=False)
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/site-packages/champ/leiden_ext.py", line 214, in parallel_leiden
    for i, res in tqdm.tqdm(enumerate(pool.imap(_run_leiden_parallel, parallel_args)), miniters=tot):
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/site-packages/tqdm/std.py", line 1180, in __iter__
    for obj in iterable:
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/multiprocessing/pool.py", line 735, in next
    raise value
TypeError: function takes at most 2 arguments (3 given)

Process finished with exit code 1
```

Other information

Ubuntu 20.04
Python 3.6 run on Pycharm

Appreciate for your assistance @wweir827

**Comment by ragibson** [comment]
This is actually a recent bug in python-igraph (see https://github.com/igraph/python-igraph/issues/454). It will be fixed in the next igraph release (0.9.9 or 0.10.0). If you need a quick fix right now, you can modify `igraph/__init__.py`, line 2315, and remove the `directed=directed, `, part from it.

There may be more issues with the test script though. I'm not sure if it was kept up to date with the rest of the project.

**Comment by balandongiv** [comment]
Thanks for dropping by @ragibson , however the problem still exist despite changes made as recommended

**Comment by ragibson** [comment]
Interesting, making the change recommended by the igraph maintainer in the linked issue resolved the problem on my end. You may be able to downgrade python-igraph as an alternative solution or wait for the fix in the next igraph release.

Do you still get the exact same error? In the traceback you posted, you can see the problem is internal to igraph at the same line that @ntamas mentioned
```python
  File "/home/rpb/anaconda3/envs/champ_test/lib/python3.6/site-packages/igraph/__init__.py", line 2315, in Read_GraphMLz
    return cls.Read_GraphML(tmpfile, directed=directed, index=index)
TypeError: function takes at most 2 arguments (3 given)
```
so editing that file should have an effect.

**Comment by ragibson** [comment]
Just checked again and the related bug in igraph is definitely fixed in the current version 0.9.10.
